src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngKey.Export.cs

```diff
@@ -44,6 +44,9 @@ internal bool TryExportKeyBlob(
             // Sanity check the current bounds
             Span<byte> empty = default;
 
+            // TODO why not just pass in destination? if it succeeds, return true,
```

Comment:
Jeremy added this like 5000 years ago, but if I had to guess it's because we want to make sure the destination length is right before we attempt to export. We don't want a partial success scenario where we pass in `destination`, but it turns out it isn't big enough, but part of the key material was written.

I suppose we could also just `Clear()` destination if it failed.

Reply:
For this PR, "because the code already exists and is well tested, and doesn't need to change".

The underlying answer is one of:
* Because this was written during a less-trusting phase when we always used the Win32 "call with NULL to get the length first" approach.
* Because some algorithm+provider pair doesn't implement export correctly with an undersized buffer, and we couldn't convert it to "try with the buffer we have, and if we get ERROR_MORE_DATA, report false".
  * That would include that some provider+algorithm pair reports something other than ERROR_MORE_DATA, and we would then grow on "not exportable" until we OOM, or we'd throw for "just try with a bigger buffer"